scene/main/viewport.cpp

```diff
@@ -50,6 +50,7 @@
 #include "scene/resources/mesh.h"
 #include "scene/scene_string_names.h"
 #include "servers/physics_2d_server.h"
+#include "editor/editor_settings.h"
```

Comment:
Any editor-specific code in `scene` should be enclosed in `#ifdef TOOLS_ENABLED`, otherwise the export template can't build.

Reply:
Actually, this should not be allowed at all as per #29730.